.github/workflows/publish-go-tester-task.yml

```
@@ -35,14 +35,21 @@ jobs:
     outputs:
       result: ${{ steps.determination.outputs.result }}
     steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
       - name: Determine if the rest of the workflow should run
         id: determination
         run: |
           RELEASE_BRANCH_REGEX="refs/heads/[0-9]+.[0-9]+.x"
+          TAG="$(git tag --points-at=HEAD 2> /dev/null | head -n1)"
           # The `create` event trigger doesn't support `branches` filters, so it's necessary to use Bash instead.
           if [[ \
-            "${{ github.event_name }}" != "create" || \
-            "${{ github.ref }}" =~ $RELEASE_BRANCH_REGEX \
+            ("${{ github.event_name }}" != "create" || \
+            "${{ github.ref }}" =~ $RELEASE_BRANCH_REGEX) && \
+            "$TAG" == "" \
```

```suggestion
            ! "${{ github.ref }}" =~ $TAG_REGEX \
```

Using `github.ref` allows significantly simplifying the workflow job by avoiding the need for the `actions/checkout` step and the complex `git tag` command.

https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push

> `ref` string Required
The full git ref that was pushed. Example: `refs/heads/main` or `refs/tags/v3.14.1`.

Demos using this approach:

- `create` event trigger for non-release branch (expected: no full run): https://github.com/per1234/arduino-cli/actions/runs/3599790649
- `create` event trigger for release branch (expected: full run): https://github.com/per1234/arduino-cli/actions/runs/3599864854
- `push` event trigger for commit that modifies relevant files (expected: full run): https://github.com/per1234/arduino-cli/actions/runs/3599873717
- `push` event trigger for tag (expected: no full run): https://github.com/per1234/arduino-cli/actions/runs/3599884119